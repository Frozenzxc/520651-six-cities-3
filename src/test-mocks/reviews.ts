import {Review} from "../types";

export const reviews: Review[] = [
  {
    comment: `What an amazing view! The house is stunning and in an amazing location. The large glass wall had an amazing view of the river!`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 1,
    rating: 4,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/8.jpg`,
      id: 17,
      isPro: false,
      name: `Emely`
    }
  },
  {
    comment: `The house is very good, very happy, hygienic and simple living conditions around it are also very good. I hope to have the opportunity to come back. Thank you.`,
    date: `2020-02-23T08:36:26.433Z`,
    id: 2,
    rating: 3,
    user: {
      avatar: `https://htmlacademy-react-3.appspot.com/six-cities/static/avatar/5.jpg`,
      id: 14,
      isPro: true,
      name: `Corey`
    }
  }
];
