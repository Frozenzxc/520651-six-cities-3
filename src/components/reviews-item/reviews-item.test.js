import React from 'react';
import renderer from 'react-test-renderer';
import reviews from "../../test-mocks/reviews";
import ReviewsItem from "./reviews-item";


it(`ReviewsItem is rendered correctly`, () => {
  const tree = renderer.create(
      <ReviewsItem
        review={reviews[0]}
      />
  ).toJSON();

  expect(tree).toMatchSnapshot();
});
