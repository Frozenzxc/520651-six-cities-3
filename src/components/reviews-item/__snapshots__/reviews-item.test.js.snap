// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReviewsItem is rendered correctly 1`] = `
<li
  className="reviews__item"
>
  <div
    className="reviews__user user"
  >
    <div
      className="reviews__avatar-wrapper user__avatar-wrapper"
    >
      <img
        alt="Reviews avatar"
        className="reviews__avatar user__avatar"
        height="54"
        src="img/avatar-max.jpg"
        width="54"
      />
    </div>
    <span
      className="reviews__user-name"
    >
      Max
    </span>
  </div>
  <div
    className="reviews__info"
  >
    <div
      className="reviews__rating rating"
    >
      <div
        className="reviews__stars rating__stars"
      >
        <span
          style={
            Object {
              "width": "20%",
            }
          }
        />
        <span
          className="visually-hidden"
        >
          Rating
        </span>
      </div>
    </div>
    <p
      className="reviews__text"
    >
      Lorem ipsum dolor sit amet, consectetur adipiscing elit. Cras aliquet varius magna, non porta ligula feugiat eget. Fusce tristique felis at fermentum pharetra. Aliquam id orci ut lectus varius viverra.
    </p>
    <time
      className="reviews__time"
      dateTime="2020-02-22"
    >
      February 22, 2020
    </time>
  </div>
</li>
`;
